### PR TITLE
Implement clear N entities performance metric

### PR DESCRIPTION
**Pull request by ianull**
To understand the overhead of clearing 0, 100, 500, ..., N entities.

### CODE REVIEW

File: javatests/arcs/android/systemhealth/testapp/StorageCore.kt [hunk 1]
```diff
@@ -350,6 +366,29 @@ class StorageCore(val context: Context, val lifecycle: Lifecycle) {
         watchdog.monitor()
     }
 
+    @Suppress("UNCHECKED_CAST")
+    private suspend fun setUpCleanerHandle(taskHandle: TaskHandle, settings: Settings) {
+        val handle = taskHandle.handleManager.createHandle(
+            HandleSpec(
+                "CleanerHandle",
+                HandleMode.ReadWrite,
```
Comment (galganif) [inline javatests/arcs/android/systemhealth/testapp/StorageCore.kt:374]: can this be HandleMode.Write?
Myk is planning some performance improvements for write only handle, so using HandleMode.Write only could result in different performances.

File: javatests/arcs/android/systemhealth/testapp/StorageCore.kt [hunk 2]
```diff
@@ -545,6 +584,36 @@ class StorageCore(val context: Context, val lifecycle: Lifecycle) {
...
+
+        if (entities.isNotEmpty()) {
+            withContext(handle.dispatcher) {
+                entities.map { handle.store(it) }
+            }.joinAll()
+            handle.getProxy().waitForIdle()
+        }
+
+        val elapsedTime = measureTimeMillis {
+            withContext(handle.dispatcher) { handle.clear() }.join()
```
Comment (galganif) [inline javatests/arcs/android/systemhealth/testapp/StorageCore.kt:605]: is joining appropriate here? (and in writerTask as well)
what does join here waits for? for the op to get to storage service?
I think clients of the handle class can use it without joining and it still should work fine

File: javatests/arcs/android/systemhealth/testapp/StorageCore.kt (hunk 1 above)
Comment (ianull) [reply javatests/arcs/android/systemhealth/testapp/StorageCore.kt:374]: Sound good! Will do

File: javatests/arcs/android/systemhealth/testapp/StorageCore.kt (hunk 2 above)
Comment (ianull) [reply javatests/arcs/android/systemhealth/testapp/StorageCore.kt:605]: Yes, we join for store, clear, client apis to wait for the op gets sent to outgoing channel and handled at ServiceStore's onProxyMessage. I did this by also referring to our codebase i.e. `dispatchStore` at HandleHelper.kt.

Comment (galganif) [reply javatests/arcs/android/systemhealth/testapp/StorageCore.kt:605]: note that dispatchStore is a testutil, not to be used in real code. But it's true that we join() in many places, we should try remove those at some point. in the meantime, maybe drop a comment here to explain why we join

File: javatests/arcs/android/systemhealth/testapp/StorageCore.kt (hunk 1 above)
Comment (ianull) [reply javatests/arcs/android/systemhealth/testapp/StorageCore.kt]: done

File: javatests/arcs/android/systemhealth/testapp/StorageCore.kt (hunk 2 above)
Comment (ianull) [reply javatests/arcs/android/systemhealth/testapp/StorageCore.kt:607]: done

**Pull request merged**